File: Data Structures & Algorithms/design-hashset/submission-0.go
type MyHashSet struct {
	m map[int]bool
}

func Constructor() MyHashSet {
    return MyHashSet{
		m: map[int]bool{},
	}
}

func (this *MyHashSet) Add(key int) {
    this.m[key] = true
}

func (this *MyHashSet) Remove(key int) {
	this.m[key] = false
}

func (this *MyHashSet) Contains(key int) bool {
	v, ok := this.m[key]
	if ok && v == true{
		return true
	}	
	return false
    
}

/**
 * Your MyHashSet object will be instantiated and called as such:
 * obj := Constructor();
 * obj.Add(key);
 * obj.Remove(key);
 * param_3 := obj.Contains(key);
 */
 

File: Data Structures & Algorithms/range-sum-query-2d-immutable/submission-0.py
class NumMatrix:

    def __init__(self, matrix: List[List[int]]):
        self.pmat = [[0 for _ in range(len(matrix[0])+1)] for _ in range(len(matrix) + 1)]

        # sum rows
        for i in range(1, len(self.pmat)):
            ps = 0
            for j in range(1, len(self.pmat[0])):
                temp = matrix[i-1][j-1]
                self.pmat[i][j] = ps + temp
                ps = ps + temp

        for j in range(1, len(self.pmat[0])):
            ps = 0
            for i in range(1, len(self.pmat)):
                temp = self.pmat[i][j]
                self.pmat[i][j] = ps + temp
                ps = ps + temp

    def sumRegion(self, row1: int, col1: int, row2: int, col2: int) -> int:
        return self.pmat[row2+1][col2+1] - self.pmat[row2+1][col1] - self.pmat[row1][col2+1] + self.pmat[row1][col1]
        


# Your NumMatrix object will be instantiated and called as such:
# obj = NumMatrix(matrix)
# param_1 = obj.sumRegion(row1,col1,row2,col2)